whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/task/grs/GrsSourceImporter.java

```diff
@@ -71,198 +71,227 @@ void grsImport(final GrsSource grsSource, final boolean rebuild) {
-                            typeAttribute.validateSyntax(cleanObject.getType(), messages);
-                            if (messages.hasErrors()) {
-                                logger.info("Errors for object with key {}: {}", typeAttribute, messages);
-                                nrIgnored++;
-                            } else if (authoritativeData.isMaintainedInRirSpace(cleanObject)) {
-                                createOrUpdate(cleanObject);
-                            }
-                        }
-                    }
+            private RpslObject filterObject(final RpslObject rpslObject) {
+                final ObjectTemplate objectTemplate = ObjectTemplate.getTemplate(rpslObject.getType());
 
-                    private RpslObject filterObject(final RpslObject rpslObject) {
-                        final ObjectTemplate objectTemplate = ObjectTemplate.getTemplate(rpslObject.getType());
+                final RpslObjectBuilder builder = new RpslObjectBuilder(rpslObject);
 
-                        final RpslObjectBuilder builder = new RpslObjectBuilder(rpslObject);
+                for (int i = 0; i < builder.size(); i++) {
+                    final RpslAttribute rpslAttribute = builder.get(i);
+                    final AttributeType attributeType = rpslAttribute.getType();
 
-                        for (int i = 0; i < builder.size(); i++) {
-                            final RpslAttribute rpslAttribute = builder.get(i);
-                            final AttributeType attributeType = rpslAttribute.getType();
+                    if (attributeType == null || !objectTemplate.hasAttribute(attributeType)) {
+                        logger.debug("Ignoring attribute in object {}: {}", rpslObject.getFormattedKey(), rpslAttribute);
+                        builder.remove(i--);
 
-                            if (attributeType == null || !objectTemplate.hasAttribute(attributeType)) {
-                                logger.debug("Ignoring attribute in object {}: {}", rpslObject.getFormattedKey(), rpslAttribute);
-                                builder.remove(i--);
+                    } else  if (attributeType.equals(AttributeType.SOURCE)) {
+                        builder.remove(i--);
+                    }
+                }
 
-                            } else  if (attributeType.equals(AttributeType.SOURCE)) {
-                                builder.remove(i--);
-                            }
-                        }
+                // best not to sort to avoid reordering remarks: attributes
+                builder.append(sourceAttribute);
 
-                        // best not to sort to avoid reordering remarks: attributes
-                        builder.append(sourceAttribute);
+                return builder.get();
+            }
 
-                        return builder.get();
-                    }
+            @Transactional
```

Comment:
@MiguelAHM I tested with the Spring transaction manager and confirmed there *is* an active transaction inside this method, i.e.
```
                final boolean actualTransactionActive = TransactionSynchronizationManager.isActualTransactionActive();
                LOGGER.info("Creating or updating object: transaction = {}", actualTransactionActive);
```

So I think we're good.

Reply:
probably you can even delete that @Transactional.
Anyways, if there is a transaction I will approve it.